packages/cli/src/helpers.ts

```diff
@@ -0,0 +1,98 @@
+import os from 'os';
+import { exec, spawnSync } from 'child_process';
+import prompts from 'prompts';
+import { magentaBright } from 'chalk';
+
+export async function setupAnvil(): Promise<void> {
+  const needsCargo = !(await hasCargo());
+  if (needsCargo) {
+    const cargoResponse = await prompts({
+      type: 'confirm',
+      name: 'confirmation',
+      message: 'Cannon requires Cargo (https://doc.rust-lang.org/cargo/). Install it now?',
+      initial: true,
+    });
+
+    if (cargoResponse.confirmation) {
+      console.log(magentaBright('Installing Cargo...'));
+      await execPromise('curl https://sh.rustup.rs -sSf | sh');
+    } else {
+      process.exit();
+    }
+  }
+
+  const MINIMUM_NIGHTLY_ANVIL_BUILD_TIMESTAMP = 1657679573421;
```

Comment:
yay, though you can probably truncate that down to a second and add a comment clarifying what date we are referring to here